### ISSUE TITLE
Not compatible with OS X Sierra

### ISSUE THREAD

**Issue by frijj2k** [mgorbach/macfusion2#52]
Macfusion will not run on OS X Sierra stating it is incompatible with this version of OS X.

**Comment by goose121** [comment]
Yeah, it won't REALLY even run on Mavericks. It doesn't let you add servers and has an exception when it starts. But the last commit was in 2010, so I think it might be abandoned.

**Comment by eastcoastcoder** [comment]
Please use ElDeveloper's fork. I'm having the same issues but at least this build has been worked on recently. https://github.com/ElDeveloper/macfusion2
Edit: Opened the same issue for you on that fork https://github.com/ElDeveloper/macfusion2/issues/17

**Comment by mmerchant** [comment]
+1 when trying on 10.12 (macOS Sierra)

> You can’t use this version of the application “Macfusion” with this version of macOS.

**Comment by luntegg** [comment]
+1

**Comment by dango** [comment]
+1

**Comment by eastcoastcoder** [comment]
Hey guys, ElDeveloper has almost got this problem taken care of. Take a look here https://github.com/ElDeveloper/macfusion2/issues/11
There is a patched build linked in the discussion. I haven't given it a spin yet but please do try it out.

**Comment by daslicht** [comment]
@goose121:
it is working fine here on OS X 10.11.6

**Comment by daslicht** [comment]
+1

**Comment by existme** [comment]
+1

**Comment by jowkar** [comment]
The patched build mentioned above runs for me, but only after also updating to the latest version of "FUSE for macOS". Prior to that it crashed instantly on start.

**Comment by fsigworth** [comment]
I LOVE MacFusion!  I use it for previewing cryo-EM images and other data on our university cluster.  It would sure be great to be able to use it again on Sierra.  Could someone point me to where I can download the latest built app?

**Comment by jankkm** [comment]
As mentioned before in this issue there is a pre release on a different fork:
https://github.com/ElDeveloper/macfusion2/releases

**Comment by jarrettj** [comment]
+1

**Comment by perdacherMartin** [comment]
I've solved the following error: "Could not mount filesystem: Mount path could not be created." by  changing the mount path under "Mount" -> "Macfusion" -> "Mount Point" to "/Users/myuser/Desktop/myHost" -> it worked.

OS: 10.12.1 and "Macfusion Sierra Pre-release 3"

**Comment by fsigworth** [comment]
So sorry to bother you-all.  I did as you said, but now get a different error, “Could not mount filesystem: Mount process terminated unexpectedly.”  In a finder window I see a folder with my host’s name appear and then in a fraction of a second disappear again, before the error message appears.

Fred

On Oct 27, 2016, at 8:50 AM, Perdacher Martin <notifications@github.com<mailto:notifications@github.com>> wrote:

I've solved the following error: "Could not mount filesystem: Mount path could not be created." by changing the mount path under "Mount" -> "Macfusion" -> "Mount Point" to "/Users/myuser/Desktop/myHost" -> it worked.

**Comment by daslicht** [comment]
`changing the mount path under "Mount" -> "Macfusion" -> "Mount Point" to "/Users/myuser/Desktop/myHost" -> it worked.`
did not work for me either :(

**Comment by AgUsBF** [comment]
hi, I am totally new at this, so excuse me if i'm asking something silly.
Yesterday I downloaded the pre-release3 and changed the mount path, worked on Sierra 10.12.1. I mounted and unmounted several times, copied and pasted files and folders... everything OK. But today when I opened macfusion the volume disappeared and I can't add a new one. The "plus" button of the app doesn't work at all, the gear button seems to work ok.
I can connect to the volume by the command ssh in the terminal, so that's ok....
Ideas?

**Comment by michaelogrant** [comment]
I have about 20 server configurations and only some of them gives me the _"Could not mount filesystem: Mount path could not be created."_ error message.  When I manipulate those problem servers' **Mount Point** setting to a read/write-able location such as `/Users/<username>/<hostname>` they mount without issue... in `/Volumes` with permissions `drwx------` and in my user directory with permissions `drwxr-xr-x@`
@goose121 : I was running the original MacFusion with recommended FUSE updates and had no major issues with it from Mavericks all the way up to and including El Capitain.  It is only with Sierra that it would no longer run.

@daslicht : Have you tried updating to the latest version of Fuse?  I'm currently still running 2.6.4... I'm hesitant to upgrade just in case it breaks!

**Comment by xxblx** [comment]
> changing the mount path under "Mount" -> "Macfusion" -> "Mount Point" to "/Users/myuser/Desktop/myHost" -> it worked.
> 
> did not work for me either :(

@daslicht see https://github.com/osxfuse/osxfuse/wiki/SSHFS#macfusion

After replacing sshfs-static mounting should works. Note that path in cd need to be replaced with actual /Applications/Macfusion\ 2.app/Contents/PlugIns/sshfs.mfplugin/Contents/Resources

**Comment by daslicht** [comment]
Still doesn't work, I have this now : 

```
drwxr-xr-x@ 10 daslicht  staff      340 Nov 27 17:02 .
drwxr-xr-x@  5 daslicht  staff      170 Oct 17 06:29 ..
-rwxr-xr-x@  1 daslicht  staff    32332 Oct 17 06:29 new_sshfs_askpass
lrwxr-xr-x   1 daslicht  staff       20 Nov 27 17:02 sshfs-static -> /usr/local/bin/sshfs
-rwxr-xr-x@  1 daslicht  staff  5845152 Oct 17 06:29 sshfs-static-lions
-rwxr-xr-x@  1 daslicht  staff  2756952 Oct 17 06:29 sshfs-static.orig
-rw-r--r--@  1 daslicht  staff    13259 Oct 17 06:29 sshfs.png
-rw-r--r--@  1 daslicht  staff     8408 Oct 17 06:29 sshfsAdvanced.nib
-rw-r--r--@  1 daslicht  staff     6987 Oct 17 06:29 sshfsConfiguration.nib
-rw-r--r--@  1 daslicht  staff    61418 Oct 17 06:29 sshfs_icon.icns
```

When I try to mount something, it just shows mounted and unmount right away.

If I remove the mount pint I still get : 
```
Could not mount filesystem: Remote host has disconnected.
```

**Comment by daslicht** [comment]
@michaelogrant :
I tried this: https://github.com/ElDeveloper/macfusion2/releases/download/2.1.1-dev.3/Macfusion.zip

**Comment by zhew117** [comment]
+1

**Comment by ychenbioinfo** [comment]
Have problem on 10.12.1 when I tried to mount a SFTP drive.
`Could not mount filesystem: Mount path could not be created.`

**Comment by fietstouring** [comment]
MacFusion worked for me on all versions of OSX until 10.11 
Just had the problem that I it always crashed when trying to delete a server in the list.
The above version has the bug fixed! Thx to the developer!

**Comment by daslicht** [comment]
Still not working here on Sierra 
(2.1.1-dev.3)

**Comment by eastcoastcoder** [comment]
Here's the way I got it working. Pay attention because you might have to try these steps MULTIPLE TIMES, as the mount point gets deleted if you have invalid credentials, or if the planets are not aligned, etc.

1. Grab ElDeveloper's Pre-Release Fork: https://github.com/ElDeveloper/macfusion2/releases
2. Make a directory in your home folder: `mkdir $HOME/Volumes/Anything`
3. Add an entry FTPFS/SSHFS
4. Change `Path: /` (Possibly mandatory for FTP)
5. Hit the Macfusion Tab
6. Change `Mount Point: /Users/USERNAME/Volumes/Anything`
7. Check the three boxes just for good measure

The `Anything` folder is a bit silly, the name will be overwritten by the provided `Volume Name` if you set one, else it will be overwritten by the `Host` url.

**Comment by daslicht** [comment]
@ethanx94 
I just tried what you suggested.
But I still get ```Could not mount filesystem: Remote host has disconnected.```

The log shows:
```
(SSHServerFS, Breb Full, 01/01/2017, 18:25) SSHFS version 2.5
(MFClient, Breb Full, 01/01/2017, 18:25) Note status changed for fs <MFClientFS: 0x6080000e9f80> (Breb Full) to Failed to Mount
(SSHServerFS, Breb Full, 01/01/2017, 18:25) mount_osxfuse: mount point /Users/daslicht/Volumes/Anything is itself on a OSXFUSE volume

```

Any idea what else to try ?

**Comment by eastcoastcoder** [comment]
That's too bad. :-(

Can you verify standard terminal SSH commands work? It seems like maybe a server security issue. Maybe too short of a timeout or the server is expecting key-based auth only.

`$ ssh username@host`

**Comment by daslicht** [comment]
yeh standard work great, I am trying this right now:

https://codingweb.io/mount-point-xyz-osxfuse-volume

But now I get an authetification error

**Comment by daslicht** [comment]
Mounting with ```sshfs``` in cli works as well but when you close the connection the Anything folder gets deleted and you have to recreate it each time ?

**Comment by eastcoastcoder** [comment]
I've noticed if your login credentials are wrong it seems to delete itself for me. I have no clue why though. If I had to speculate, it could be trying to overwrite the `Anything` folder with the provided `Volume Name` but then realizing your login stuff is wrong, it stops midway and doesn't replace the folder with `Volume Name` but just deletes it completely. Haha.

As far as your situation, that's really strange. If it helps I'm running Sierra 10.12.1, with FUSE for macOS 3.5.4. 

`mount point /Users/daslicht/Volumes/Anything is itself on a OSXFUSE volume` is a weird error. It's mentioned here though, so if you really wanna poke at it you might try some of the things posted here. https://github.com/osxfuse/osxfuse/issues/45#issuecomment-47611856

**Comment by daslicht** [comment]
Thats currently the only software which stops me from completely upgrading to Sierra :/

**Comment by daslicht** [comment]
Thats interesting: 

```
debug1: Authentications that can continue: publickey,password
debug1: Next authentication method: publickey
debug1: Trying private key: /Users/daslicht/.ssh/id_rsa
debug1: Trying private key: /Users/daslicht/.ssh/id_dsa
debug1: Trying private key: /Users/daslicht/.ssh/id_ecdsa
debug1: Trying private key: /Users/daslicht/.ssh/id_ed25519
debug1: Next authentication method: password
debug1: read_passphrase: can't open /dev/tty: Device not configured

```

**can't open /dev/tty: Device not configured**

**Comment by daslicht** [comment]
I have created a OS X Command which lets you mound remote volumes with a double click: 

https://github.com/daslicht/mount-sshfs

**Comment by lebigot** [comment]
Thank @ethanx94, your method worked for me. The `mkdir` command in your method would be better as `mkdir -p …` (the option is necessary, for the default `mkdir`).

**Comment by fleXible** [comment]
+1

**Comment by lebigot** [comment]
Now getting, on a different machine:

```
(SSHServerFS, ***, 2017-01-16, 09:32) dyld: Library not loaded: /usr/local/lib/libfuse.0.dylib

  Referenced from: /Applications/Macfusion.app/Contents/PlugIns/sshfs.mfplugin/Contents/Resources/sshfs-static

  Reason: image not found

```

Any idea of how this could be solved? Scouring the internet did not give me a solution (including trying `ln -s /usr/local/lib/libosxfuse.2.dylib /usr/local/lib/libfuse.0.dylib`).

**Comment by anignatev** [comment]
Mac OS Sierra 10.12.2 (16C68), Macfusion Sierra Pre-release 3

EncFS plugin is not work ((

When push the button "Mount", displaying error: Decryption failed: wrong password

sudo dmesg:
AMFI: allowing exception handler for 'encfs_askpass' (31515) because the process is unsigned.

**Comment by ximm** [comment]
@lebigot with Sierra 10.12.3 I was able to successfully mount a volume with sshfs after linking with the `ln -s` as you show... 

...but then would get permanent hanging (until unmounting).

But just now I was able to successfully mount and access my remote, after I realized I had overlooked this note on the macfusion2 repo readme, about the compatibility layer in osxfuse being required: 

> For Mac OS X 10.7 and greater, you need Fuse for OSX, note that you will need to check the "install MacFUSE compatibility layer" option during the installation process, otherwise some parts of the software will be broken.

For me the key was installing the compatibility layer :P

(I have osxfuse 3.5.5)

**Comment by lebigot** [comment]
Thanks @ximm: enabling the compatibility layer worked for me too! After all these years I naively thought that all basic features had been updated. :)

**Comment by nilx** [comment]
can't run on Sierra 10.12.15

```
dyld: Library not loaded: @rpath/MFCore.framework/Versions/A/MFCore
  Referenced from: /Applications/Macfusion.app/Contents/MacOS/Macfusion
  Reason: no suitable image found.  Did find:
	/Applications/Macfusion.app/Contents/MacOS/../Frameworks/MFCore.framework/Versions/A/MFCore: cannot load '/Applications/Macfusion.app/Contents/MacOS/../Frameworks/MFCore.framework/Versions/A/MFCore' because Objective-C garbage collection is not supported
Abort trap: 6
```

**Comment by bharatkrishna** [comment]
I'm still getting "Could not mount filesystem: Remote host has disconnected." after follows the steps in this thread. Any workarounds?

**Comment by Relequestual** [comment]
I had to try https://github.com/mgorbach/macfusion2/issues/52#issuecomment-269890019 multiple times, and reboot between, before it would work.

**Comment by luntegg** [comment]
@daslicht @bharatkrishna 

I had the same problem – "Could not mount filesystem: Remote host has disconnected.". After reading all posts here, I tried to connect using mac terminal and it's failed because password was incorrect. After restoring access to the server and convinced that my new password correct (in terminal), I was able to mount the volume using macfusion.

**Comment by bharatkrishna** [comment]
In my organization, we were recently asked to add a HostKeyAlgorithm to ssh (`ssh -oHostKeyAlgorithms=+ssh-dss`). If I don't add this, ssh fails. 

Adding this to sshfs (`sshfs -oHostKeyAlgorithms=+ssh-dss`) I'm able to mount the filesystem via cli but doesn't work when I try it via MacFusion. I added it in the 'SSH Advanced' tab in 'Extra Options', but didn't work. Now it says "Could not mount filesystem: Mount process has terminated unexpectedly."

**Comment by pdwalker** [comment]
Does anyone have a working pre-compiled binary?

**Comment by cukabeka** [comment]
Yes, I'm looking for a compiled App, too. I tried the releases here https://github.com/ElDeveloper/macfusion2/releases but could only find the sourcecode. I would so appreciate if you guys would merge ElDeveloper's work here and release a new binary!
Has anybody got this running on Catalina, BTW?
